handler.go

```diff
@@ -181,11 +185,14 @@ func (h *socketHandler) onPacket(decoder *decoder, packet *packet) ([]interface{
 }
 
 func (h *socketHandler) onAck(id int, decoder *decoder, packet *packet) error {
+	h.acksmu.Lock()
 	c, ok := h.acks[id]
 	if !ok {
+		h.acksmu.Unlock()
```

Comment:
Using defer would hold the lock longer than necessary since `h.acks` is modified only at the beginning of the function.